lib/templateFactory.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const BaseFactory = require('./baseFactory');
+const Template = require('./template');
+let instance;
+
+class TemplateFactory extends BaseFactory {
+    /**
+     * Construct a JobFactory object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Datastore}    config.datastore         Object that will perform operations on the datastore
+     */
+    constructor(config) {
+        super('template', config);
+    }
+
+    /**
+     * Instantiate a Template class
+     * @method createClass
+     * @param  {Object}     config               Template data
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.version       Version of the template
+     * @param  {String}     config.description   Description of the template
+     * @param  {String}     config.maintainer    Maintainer's email
+     * @param  {Object}     config.config        Config of the screwdriver-template.yaml
+     * @param  {String}     config.scmUri        ScmUri of the template pipeline
+     * @param  {Array}      config.labels        Labels attached to the template
```

Comment:
The `README` states that this is optional, but the JSDoc does not reflect that. Which one is it?/